src/main/java/com/synopsys/integration/detect/workflow/bdio/AggregateBdioCreator.java

```diff
@@ -129,13 +130,17 @@ public AggregateBdioCreator(final Bdio2Factory bdio2Factory, final SimpleBdioFac
         }
     }
 
-    private DependencyGraph createAggregateDependencyGraph(final File sourcePath, final List<DetectCodeLocation> codeLocations) {
+    private DependencyGraph createAggregateDependencyGraph(final File sourcePath, final List<DetectCodeLocation> codeLocations, final AggregateMode aggregateMode) {
         final MutableDependencyGraph aggregateDependencyGraph = simpleBdioFactory.createMutableDependencyGraph();
 
         for (final DetectCodeLocation detectCodeLocation : codeLocations) {
-            final Dependency codeLocationDependency = createAggregateDependency(sourcePath, detectCodeLocation);
-            aggregateDependencyGraph.addChildrenToRoot(codeLocationDependency);
-            aggregateDependencyGraph.addGraphAsChildrenToParent(codeLocationDependency, detectCodeLocation.getDependencyGraph());
+            if (aggregateMode.equals(AggregateMode.DIRECT)) {
+                aggregateDependencyGraph.addGraphAsChildrenToRoot(detectCodeLocation.getDependencyGraph());
+            } else {
+                final Dependency codeLocationDependency = createAggregateDependency(sourcePath, detectCodeLocation);
+                aggregateDependencyGraph.addChildrenToRoot(codeLocationDependency);
+                aggregateDependencyGraph.addGraphAsChildrenToParent(codeLocationDependency, detectCodeLocation.getDependencyGraph());
+            }
```

Comment:
I'd prefer it if the "else" was an "else if" checking for the "TRANSITIVE" mode. An additional else block should be created which throws an exception or something. Since AggregateMode is an Enum and that Enum could change, having the "else" drive the functionality seems wrong. Additionally one of the arguments for using an enum was to allow for expansion in the future, which this code doesn't address.